include/bluetooth/nrf/host_extensions.h

```diff
@@ -0,0 +1,126 @@
+ *                           Disabled by default.
+ *  @param beta
+ *  @parblock
+ *  Value used to determine the weight of the previous average of RSSI values.
+ *  A higher value lowers how much the current RSSI weighs into the average,
+ *  flattening peaks, which also means the controller reacts slower
+ *  on RSSI changes.
+ *  The valid range is [0, 4095].
+ *  Default value is 2048.
+ *
+ *  In the controller, the value is used in an exponential weighted
+ *  averaging of RSSI in a 12-bit fixed point fraction.
+ *
+ *  avg[n] = gamma * avg[n - 1] + (1 - gamma) * rssi[n]
+ *
+ *  Here, gamma equals beta/4096, and rssi[n] equals the current RSSI.
+ *  For example, for gamma to be 0.25,
+ *  set the beta parameter in the command to 1024.
+ *  @endparblock
+ *  @param lower_limit       The lower limit of the RSSI golden range. The RSSI golden range
+ *                           is explained in Core_v5.4, Vol 6, Part B, Section 5.1.17.1.
+ *                           Default value is -70 dBm.
+ *  @param upper_limit       The upper limit of the RSSI golden range.
+ *                           Default value is -30 dBm.
+ *  @param lower_target_rssi Target RSSI level in dBm units when the average
+ *                           RSSI level is less than the lower limit of
+ *                           the RSSI golden range.
+ *                           Default value is -65 dBm.
+ *  @param upper_target_rssi Target RSSI level in dBm units when the average
+ *                           RSSI level is greater than the upper limit of
+ *                           the RSSI golden range.
+ *                           Default value is -35 dBm.
+ *  @param wait_period_ms    Duration in milliseconds to wait before initiating a
+ *                           new LE Power Control Request procedure by the controller.
+ *                           It is needed to not repeat send requests for transmit power change
+ *                           without the remote having had the chance to react, as well as to avoid
+ *                           a busy controller. This value should be set depending on needs.
+ *                           0 milliseconds value is an invalid value.
+ *                           Default value is 5000 milliseconds.
+ *  @param apr_margin        Margin between APR value received from peer in LL_POWER_CONTROL_RSP PDU
+ *                           and actual reduction in Transmit power that is applied locally.
+ *                           The applied decrease in local Transmit power will be
+ *                           (received_apr - apr_margin)
+ *                           if received_apr > apr_margin, otherwise no change.
+ *                           Default value is 5 dB.
+ *
+ *  @return Zero on success or (negative) error code on failure.
+ *  @retval -ENOBUFS HCI command buffer is not available.
+ */
+int bt_conn_set_power_control_request_params(uint8_t auto_enable,
```

Comment:
This function takes a lot of parameters. Would it be better to group them in a parameter struct like we do in some other places in the bluetooth subsys?